### PR TITLE
#371 #180 #175 #174 #364 #370 #369 Poblem and ProblemSet functionalities

### PR DESCRIPTION
**Pull request by afijal**
*adding new problems to problemSet
*removing problems from problemset
*view only problem view
*share modals
and so on

### CODE REVIEW

File: src/components/Editor/components/MyWork/components/MyWorkEditorButtons/components/MyWorkFooter/index.js [hunk 1]
```diff
@@ -24,24 +21,46 @@ export default class MyWorkFooter extends Component {
...
-        )
-    }
-
     render() {
         const btnClassNames = [
             bootstrap.btn,
             styles.pointer
         ];
 
+        var leftButton = this.props.addingProblem ?
```
Comment (oskarhinc) [inline src/components/Editor/components/MyWork/components/MyWorkEditorButtons/components/MyWorkFooter/index.js]: I think moving all those buttons to a variables could improve the readability

File: src/components/Editor/components/ProblemHeader/styles.css [hunk 2]
```diff
@@ -8,15 +8,30 @@
...
     margin-right: 5px;
 }
+
+.editLink {
+    margin-right: 5px;
+    min-width: 400px;
+}
+
+.editLinkLabel {
+    color: #FFF;
```
Comment (oskarhinc) [inline src/components/Editor/components/ProblemHeader/styles.css]: please use rgb colors

File: src/components/Editor/components/ProblemHeader/styles.css [hunk 3]
```diff
@@ -8,15 +8,30 @@
...
     background-color: rgb(0, 98, 217);
-    padding: 5px;
+    padding-top: 10px;
     display: flex;
-    justify-content: space-between;
 }
 
 .btnContainer {
     white-space: nowrap;
+    margin-left: auto
```
Comment (oskarhinc) [inline src/components/Editor/components/ProblemHeader/styles.css]: missing semicolon

File: src/components/Editor/index.js [hunk 4]
```diff
@@ -372,10 +386,49 @@ export default class Editor extends Component {
...
+        }
+        )
+    };
+
+    saveProblem() {
+        googleAnalytics('Save');
+        axios.put(`${config.serverUrl}/solution/`, this.state.solution)
+            .then(response => {
+                this.setState({editLink: config.serverUrl + '/problem/edit/' + this.state.solution.editCode})
+                createAlert('success', Locales.strings.problem_saved_success_message, 'Success');
```
Comment (oskarhinc) [inline src/components/Editor/index.js]: Please move "Success" to strings.js".

File: src/components/Editor/styles.css [hunk 5]
```diff
@@ -55,4 +55,3 @@
 .contentWrapper {
     display: contents;
 }
```
Comment (oskarhinc) [inline src/components/Editor/styles.css]: restore empty line at the end of the file

File: src/components/Home/components/Header/index.js [hunk 6]
```diff
@@ -8,15 +8,33 @@ import { withRouter } from 'react-router-dom';
...
     render() {
-        const Button = withRouter(({ history }) => (
+        const GettingStartedButton = withRouter(({ history }) => (
             <a className={classNames(bootstrap['nav-link'], header.pointer)} onClick={() => { history.push('/problem/example') }}>
-                Getting Started
+                {Locales.strings.getting_started_title}
             </a>
         ))
 
+        const Button = this.props.editing ?
```
Comment (oskarhinc) [inline src/components/Home/components/Header/index.js]: shouldn't those variables be lowercase?

File: src/components/Home/components/Navigation/Problem/index.js [hunk 7]
```diff
@@ -43,7 +51,8 @@ export default class Problem extends Component {
         }
         axios.post(`${config.serverUrl}/solution/new`, solution)
             .then(response => {
-                history.push('/problem/view/' + response.data.shareCode);
+                console.log(response.data)
```
Comment (oskarhinc) [inline src/components/Home/components/Navigation/Problem/index.js]: do we need this console log?

File: src/components/Home/components/Navigation/Problem/index.js [hunk 8]
```diff
@@ -53,10 +62,35 @@ export default class Problem extends Component {
...
+            className={
+                classNames(
+                    problem.plusIcon,
+                    'fa-2x'
+                )
+            }
+            name='plus-circle'
+        />
+        : null;
+        var removeButton = this.props.showRemove ?
```
Comment (oskarhinc) [inline src/components/Home/components/Navigation/Problem/index.js]: add empty line here to improve readablity

File: src/components/Home/components/NewProblemsForm/styles.css [hunk 9]
```diff
@@ -0,0 +1,26 @@
+.container {
+    width: 1100px;
+    height: 700px;
+    background: #dadada;
```
Comment (oskarhinc) [inline src/components/Home/components/NewProblemsForm/styles.css]: rgb

File: src/components/Home/components/NewProblemsForm/styles.css [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+.row, .header {
+    display: flex;
+    justify-content: space-around;
+    margin: 5px;
+}
+
+.row {
+    background: #ffffff;
```
Comment (oskarhinc) [inline src/components/Home/components/NewProblemsForm/styles.css]: rgb

File: src/components/Home/index.js [hunk 11]
```diff
@@ -10,36 +10,176 @@ import createAlert from '../../scripts/alert';
...
+        this.setState({ shareModalActive: true });
+    };
+
+    activateModal() {
+        this.setState({ modalActive: true });
+    };
+
+    addProblem() {
+        if (this.state.textAreaValue === "") {
+            createAlert('warning', Locales.strings.no_problem_title_warning, 'Warning');
```
Comment (oskarhinc) [inline src/components/Home/index.js]: Please move "Warning" to strings.js

File: src/components/Home/index.js [hunk 12]
```diff
@@ -10,36 +10,176 @@ import createAlert from '../../scripts/alert';
...
+            setTimeout(function(){
+                $('#mathAnnotation').focus();
+            },6000);
```
Comment (oskarhinc) [inline src/components/Home/index.js]: add empty space before 6000

File: src/components/Home/index.js [hunk 13]
```diff
@@ -10,36 +10,176 @@ import createAlert from '../../scripts/alert';
...
+            this.setState({
+                set: {
+                    problems: response.data.problems,
+                    editCode: response.data.editCode,
+                    sharecode: response.data.shareCode
+                },
+                tempProblems: [],
+                modalActive: false
             });
+            console.log(response.data);
```
Comment (oskarhinc) [inline src/components/Home/index.js]: is it needed here?

File: src/components/Home/index.js [hunk 14]
```diff
@@ -10,36 +10,176 @@ import createAlert from '../../scripts/alert';
...
+            this.setState({
+                set: {
+                    problems: response.data.problems,
+                    editCode: response.data.editCode,
+                    sharecode: response.data.shareCode
+                },
+                tempProblems: [],
+                modalActive: false
+            });
+            console.log(response.data);
```
Comment (oskarhinc) [inline src/components/Home/index.js]: is it needed here?

File: src/components/ShareModal/styles.css [hunk 15]
```diff
@@ -0,0 +1,34 @@
+.modal {
+    background: #fff;
```
Comment (oskarhinc) [inline src/components/ShareModal/styles.css]: rgb

File: src/components/ShareModal/styles.css [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.modalHeader,
+.modalBody,
+.modalFooter {
+    padding: 0.5em 1.5em;
+}
+
+.modalHeader {
+    border-bottom: 1px solid #eee;
```
Comment (oskarhinc) [inline src/components/ShareModal/styles.css]: rgb

File: src/components/ShareModal/styles.css [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.modalTitle {
+    margin: 0;
+}
+
+.modalFooter {
+    border-top: 1px solid #eee;
```
Comment (oskarhinc) [inline src/components/ShareModal/styles.css]: rgb

File: src/strings.js [hunk 18]
```diff
@@ -68,7 +72,23 @@ class Locales {
...
+            add_problem_title: "Add new Problem",
+            add_problem: " Add Problem",
+            add_problem_equation: "+",
+            save: "Save",
+            save_button: " Save",
+            edit_link_label: "Edit Url: ",
+            cancel: " Cancel",
+            new_problem: " New Problem",
+            add_problems: " Add problem(s) to set",
+            hash: "#",
```
Comment (oskarhinc) [inline src/strings.js]: I don't think hash is needed to be in strings.js

File: src/components/Editor/index.js [hunk 19]
```diff
@@ -372,10 +386,49 @@ export default class Editor extends Component {
         this.scrollToBottom();
     }
 
+    getProblemById(problems, id) {
```
Comment (oskarhinc) [inline src/components/Editor/index.js]: is it used anywhere?

File: src/components/Editor/styles.css (hunk 5 above)
Comment (afijal) [reply src/components/Editor/styles.css]: there is one, it was doubled :)

File: src/strings.js (hunk 18 above)
Comment (afijal) [reply src/strings.js]: well technically it is locale dependant, at least I think so

File: src/components/Editor/styles.css (hunk 5 above)
Comment (oskarhinc) [reply src/components/Editor/styles.css]: cool

**Pull request merged**